packages/plugins/i18n/admin/src/components/CMHeaderActions.tsx

```diff
@@ -226,6 +237,299 @@ const DeleteLocaleAction: DocumentActionComponent = ({
+};
+
+const BulkLocalePublishAction: DocumentActionComponent = ({
+  document: baseDocument,
+  documentId,
+  model,
+  collectionType,
+}) => {
+  const baseLocale = baseDocument?.locale ?? null;
+
+  const [{ query }] = useQueryParams<{ status: 'draft' | 'published' }>();
+
+  const params = React.useMemo(() => buildValidParams(query), [query]);
+  const isPublishedTab = query.status === 'published';
+
+  const { formatMessage } = useIntl();
+  const { hasI18n, canPublish } = useI18n();
+  const { toggleNotification } = useNotification();
+  const { _unstableFormatAPIError: formatAPIError } = useAPIErrorHandler();
+
+  const [selectedRows, setSelectedRows] = React.useState<any[]>([]);
+  const [isConfirmationOpen, setIsConfirmationOpen] = React.useState<boolean>(false);
+
+  const { publishMany: publishManyAction } = useDocumentActions();
+  const {
+    document,
+    meta: documentMeta,
+    schema,
+    validate,
+  } = useDocument({
+    model,
+    collectionType,
+    documentId,
+    params: {
+      locale: baseLocale,
+    },
+  });
+
+  const { data: localesMetadata = [] } = useGetLocalesQuery();
+
+  const headers = [
+    {
+      label: formatMessage({
+        id: 'global.name',
+        defaultMessage: 'Name',
+      }),
+      name: 'name',
+    },
+    {
+      label: formatMessage({
+        id: getTranslation('CMEditViewBulkLocale.status'),
+        defaultMessage: 'Status',
+      }),
+      name: 'status',
+    },
+    {
+      label: formatMessage({
+        id: getTranslation('CMEditViewBulkLocale.publication-status'),
+        defaultMessage: 'Publication Status',
+      }),
+      name: 'publication-status',
+    },
+  ];
+
+  const [rows, validationErrors] = React.useMemo(() => {
+    if (!document || !documentMeta?.availableLocales) {
+      return [[], {}];
+    }
+
+    const rowsFromMeta: LocaleStatus[] = documentMeta?.availableLocales.map((doc) => {
+      const { locale, status } = doc;
+
+      return { locale, status };
+    });
+    rowsFromMeta.unshift({
+      locale: document?.locale ?? '',
+      status: document?.status ?? 'draft',
+    });
+
+    const allDocuments = [document, ...(documentMeta?.availableLocales ?? [])];
+
+    const errors: Record<
+      Modules.Documents.Params.Locale.StringNotation,
+      Record<string, MessageDescriptor>
+    > = {};
+
+    allDocuments.map((document) => {
+      if (!document) {
+        return document;
+      }
+
+      const validation = validate(document as Modules.Documents.AnyDocument);
+      if (validation !== null) {
+        errors[document.locale] = validation;
+      }
+
+      return document;
+    });
+
+    return [rowsFromMeta, errors];
+  }, [document, documentMeta?.availableLocales, validate]);
+
+  const localesToPublish = selectedRows
+    .filter(
+      (selectedRow) =>
+        selectedRow.status !== 'published' &&
+        !Object.keys(validationErrors || {}).includes(selectedRow.locale)
```

Comment:
```suggestion
        !Object.keys(validationErrors).includes(selectedRow.locale)
```

validationErrors is always an object i think?